frontend/src/pages/Reservation.tsx

```diff
@@ -49,6 +53,33 @@ function Reservation() {
     setReservation(reservationQuery.data);
   }, [token]);
 
+  useEffect(()=>{
+    refetchActivities();
+  }, [roomId])
+
+  let getRoomActivities = () => {
+    if(roomId === undefined) return [];
+    return Api.Api.get(`activity/room/${roomId}`).then((res) => res.data);
```

Comment:
Do we want to return all activites for all days when we only want to display single day in calendar? I think we should have another endpoint for this.

Reply:
That really depends. Even though we only display a narrow view of just a single day, we can still allow the user to move the calendar view forwards and backwards for a better perspectve. In this scenario, fetching all activities is probably fine and a more optimized alternative would be to have a partial fetch of just neighboring dates (which I argue would deserve it's own task).

We could also, as you say, strip this functionality and only fetch data for a single day, which I think would be a net loss for the user in exchange for a minor optimization. What do you think?

Also, in case we go with the first option, I just found out that "Dziś"/"Następny"/"Poprzedni" buttons don't actually  work, so I'll put a fix for them.